// SCL (or protobuf) representation of an Iris radio
package scl_radioconfig;

enum ControlType {
  REQUEST = 1;
  REPLY = 2;
}

enum CommandId {
  GET_RADIO_CONFIG = 1;
  SET_RADIO_CONFIG = 2;
}

message ComponentParameter {
  required string name = 1;
  required string value = 2;
}

message RadioComponent {
  required string name = 1;
  repeated ComponentParameter parameters = 2;
}

message RadioEngine {
  required string name = 1;
  repeated RadioComponent components = 2;
}

message RadioConfig {
  repeated RadioEngine engines = 1;
}

message RadioConfigControl {
  required ControlType type = 1;
  required CommandId command = 2;
  optional RadioConfig radioconf = 3;
}
